### DISCUSSION TITLE
Yarn v4 support

### DISCUSSION THREAD

**Discussion by anthony-hayes** [vercel/turborepo#7931] [Help] [answered]
### Summary

In TurboRepo's [support policy](https://turbo.build/repo/docs/support#package-managers), it says the following
* yarn (v1+) (v4 partially supported)

What does "partially supported" mean? What are the things that are not supported? 

It'd be nice to have a tag on yarn 4 issues blocking this, so as a community we can better understand where progress on this is. (perhaps a tag per package manager for issues might help)

Yarn v4 came out in October 23, 2023, and I'm eager to upgrade, but it's not something that's easily rolled back. Hence some clarity would be wonderful.

Thanks for the work on TurboRepo, it's an excellent tool.

**Comment by anthonyshew** [comment, marked as answer]
The Support Policy currently says "partially supported" because we haven't had many folks use v4 with Turborepo. As you can imagine, there are _many_ permutations of monorepos out there and we don't want to promise true stability for something we aren't confident in.

I can say from past experience, the lacking support is usually found in the `prune` command. You're certainly welcome to give things a try in a low risk environment, if you have one. We're happy to hear your feedback so we can get closer to being confident enough to mark it as full support.

**Reply by anthony-hayes** [reply]
Ah that's a shame, the `prune` command is exactly what I'm looking for. Thanks for the information!

**Reply by anthonyshew** [reply]
It may work in your repo so I'd suggest giving it a try. Like I said, there are lots of different things package managers do/can do so getting everything right is a matter of folks using it and letting us know what doesn't work so we can fix it.

If you can try it out and let us know how things go, we'd love an Issue report with as minimal of a reproduction as possible for anything you may come across.

**Reply by anthony-hayes** [reply]
Anecdotally it worked, but with something as complex as `prune` I think I'd want a bit more confidence for high-risk projects.
My understanding is there's not much changes between v3 and v4 of yarn in regards lockfiles.
I guess there's a bit of a chicken and egg with partial support - folks might not move unless they're confident in the tool.
Perhaps getting a changelog for the yarn.lock format changes would help? The yarn discord is very supportive.

**Reply by anthony-hayes** [reply]
I'd love to test it out more, but it's hard to justify on larger high-risk projects, and prune issues are (I'm assuming) very difficult to identify and reproduce.

It might be worth creating an issue to track the status of v4 support, and what you folks would need to get it to full support. Automated tests, etc etc. I'd love to help out in my spare time
